lib/fs.js

```diff
@@ -1617,8 +1617,15 @@ function ReadStream(path, options) {
   if (!(this instanceof ReadStream))
     return new ReadStream(path, options);
 
+  if (options === undefined)
```

Comment:
hm. OK. I will disallow `null`. 

our current spec is to allow `object` only.
And under this PR, we can use `string` as an encoding. 

So, this function allows `object` and `string`. the other types are not allowed.
If we found disallowed type, we should throw `TypeError`.

But null is unclear type.
I think null and undefined are not easy to use properly for beginner. 
So my first impression, we should allow null just like undefined. 
But I will follow @cjihrig suggestion. his spec is clear.